mb_add_remix_credit_links.user.js

```diff
@@ -143,48 +138,53 @@ function addRemixCreditClickHandler(event) {
     const remixer = this.getAttribute('data-remixer');
     const remixOf = this.getAttribute('data-remix-of');
 
-    const addRel = recording.getElementsByClassName('add-rel')[0];
+    const addRel = recording.querySelector('button.add-relationship');
     addRel.click();
 
     if (remixer) {
         // wait 250ms for the dialog to be added to the DOM
         window.setTimeout(function () {
-            const dialog = document.getElementById('dialog');
-            const linkType = dialog.getElementsByClassName('link-type')[0];
-            setElementValue(linkType, RemixerOptionValue, 'change');
+            const dialog = document.getElementById('add-relationship-dialog');
+            const entityType = dialog.querySelector('select.entity-type');
+            setElementValue(entityType, 'artist', 'change');
+
+            const linkType = dialog.querySelector('input.relationship-type');
+            setElementValue(linkType, 'remixed / remixer');
+            tabToConfirmFirstOption(linkType);
 
-            const name = dialog.getElementsByClassName('name')[0];
+            const name = dialog.querySelector('input.relationship-target');
             if (remixer) {
                 setElementValue(name, remixer);
-            } else {
-                name.focus();
             }
+            name.focus();
```

Comment:
Looking at this now I can't see how it's possible for `remixer` to be falsey so `if (remixer)` seems redundant. Let me know if I'm missing something obvious!

The following change resolved the stuck loading animation issue for me:

```js
            setElementValue(name, remixer);

            const search = name.parentElement.querySelector('button.search');
            if (search) {
                window.setTimeout(function() {
                    search.click();
                    name.focus();
                }, 100);
            }
```

Reply:
Thanks for the comments! I also encountered the search issue, but was struggling to come up with a solution. So thanks for the suggestion, it worked for me as well now that I tested it in Tampermonkey. I've pushed the fixes now.